### PR TITLE
Dashboard

### PR DESCRIPTION
**Pull request by bcorwin**
Dashboard with gauges, painted table, and update button (testing results TDB).

### CODE REVIEW

File: monitor/templates/dashboard.html [hunk 1]
```diff
@@ -0,0 +1,182 @@
+<!DOCTYPE html>
+<!-- Template by html.am -->
+
+<html>
+	<head>
+		<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
+		<title>Dashboard</title>
+		<style type="text/css">
```
Comment (cutreth) [inline monitor/templates/dashboard.html]: Take a look at what I've done in the other HTML files. We can pull the CSS out from here into one central file.

File: monitor/urls.py [hunk 2]
```diff
@@ -10,4 +10,6 @@
     url(r'^chart/(?P<cur_beer>\d+)/$', views.chart, name='chart_beer'),
     url(r'^graph/$', views.graph, name='graph_default'),
     url(r'^graph/(?P<cur_beer>\d+)/$', views.graph, name='graph_beer'),
+    url(r'^dashboard/$', views.dashboard, name='dashboard_default'),
+    url(r'^dashboard/(?P<cur_beer>\d+)/$', views.dashboard, name='graph_beer'),
```
Comment (cutreth) [inline monitor/urls.py]: I had only imagined we'd do the dashboard for the active beer. It's fine if we don't, but personally I'd just start there.

File: monitor/views.py [hunk 3]
```diff
@@ -538,3 +538,44 @@ def createFig(vers, active_beer):
...
     
+def dashboard(request, cur_beer=None):
+    ## !RANGES ONLY EXIST FOR ACTIVE BEER! ##
+    if cur_beer is None:
+        active_config = Config.objects.filter()[:1].get()
+        active_beer = active_config.beer
+    else:
+        active_beer = Beer.objects.get(pk=cur_beer)
+    # To do:
+    # -Get most recent logged values (or send2middlware("r=var")?)
```
Comment (cutreth) [inline monitor/views.py]: If you do this, we should write a new command to return all the readings at once. If we do that though, does it make more sense to just for a log each time?

File: monitor/views.py [hunk 4]
```diff
@@ -538,3 +538,44 @@ def createFig(vers, active_beer):
...
+    # -Can use setInterval js function to auto update with send2middlware("r=var")?
+    # -Add dashboard graphs
+    # -Add button to force a log and refresh page
+    # -and/or add button to send2middleware("r=var") and update charts
+    # -Add footnote of time of last log
+    # -Function to find bgcol (and fgcol)
+    # -How much time since last log
+    cur_reading = [r for r in Reading.objects.filter(beer=active_beer).order_by("-instant_actual")[:1]][0]
```
Comment (cutreth) [inline monitor/views.py]: I don't think you need the [r for r in] piece. You can probably get by with something similar to what I'm doing for the active_config look up.

File: monitor/views.py [hunk 5]
```diff
@@ -538,3 +538,44 @@ def createFig(vers, active_beer):
...
+            "pres_beer": "white"
+        },
+        "greenrng": {
+            "temp_amb": [active_config.temp_amb_base - active_config.temp_amb_dev, active_config.temp_amb_base + active_config.temp_amb_dev],
+            "temp_beer": [active_config.temp_beer_base - active_config.temp_beer_dev, active_config.temp_beer_base + active_config.temp_beer_dev]
+        },
+        "last_log_date": cur_reading.instant_actual.strftime("%Y-%m-%d"),
+        "last_log_time": cur_reading.instant_actual.strftime("%H:%M:%S"),
+        'all_beers': Beer.objects.all(),
+        'test': int(active_config.temp_amb_base - active_config.temp_amb_dev)
```
Comment (cutreth) [inline monitor/views.py]: What is this for?

File: monitor/urls.py (hunk 2 above)
Comment (bcorwin) [inline monitor/urls.py]: Yea, when I did that I didn't realize the bounds were only for active. I'll remove this functionality.

File: monitor/views.py (hunk 3 above)
Comment (bcorwin) [inline monitor/views.py]: I won't be doing this, it was just an early thought.

File: monitor/views.py (hunk 5 above)
Comment (bcorwin) [inline monitor/views.py]: Possible future functionality (painting the cells in the table) if the table stays (probably not now that I have the gauges up and running).

Comment (cutreth) [inline monitor/views.py]: Correct - 'test'.

File: monitor/urls.py [hunk 6]
```diff
@@ -10,4 +10,5 @@
     url(r'^chart/(?P<cur_beer>\d+)/$', views.chart, name='chart_beer'),
     url(r'^graph/$', views.graph, name='graph_default'),
     url(r'^graph/(?P<cur_beer>\d+)/$', views.graph, name='graph_beer'),
+    url(r'^dashboard/$', views.dashboard, name='dashboard_default'),
```
Comment (cutreth) [inline monitor/urls.py]: You use the "name" value to do a reverse URL lookup, so there shouldn't be any that are identical.

File: monitor/urls.py [hunk 7]
```diff
@@ -10,4 +10,6 @@
...
+    url(r'^dashboard/update/$', views.dashboard_update, name='dashboard_default'),
```
Comment (cutreth) [inline monitor/urls.py]: Make sure there are no duplicates "name" values.

File: monitor/views.py [hunk 8]
```diff
@@ -538,3 +538,68 @@ def createFig(vers, active_beer):
...
+    if rng == None: bgcol = "#FFFFFF" #White
+    elif(rng[0] <= val <= rng[1]): bgcol = "#008000" #Green
+    else: bgcol = "#FF0000" #Red
+    
+    fgcol = "#000000" #Black
+    return((bgcol, fgcol))
+def dashboard_update(request):
+    command_status = send2middleware("F")
+    #Add a pop-up with error status
+    return dashboard(request)
```
Comment (cutreth) [inline monitor/views.py]: This isn't exactly right. Take a look at how I handle the command functionality. You'll want to do a redirect here.

**Comment by cutreth** [comment]
Ping me when you have put your last stable commit in and we can push to PRD for testing.

**Pull request merged**